### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by MokhammadDaniyal**
**What it does ?**
 Added Profile Page + Profile Edit (Edit name, edit profile image, edit experience)
**What to avoid ?**
N/A
**What to look for ?**
any possible typos, bugs,etc.
Right now Changing experience is not working. It looks like restructuring database tables caused the problem @ko1keith  Please have a look.
Screenshots
Screen Shot 2020-11-02 at 5 11 38 PM
Screen Shot 2020-11-02 at 5 11 48 PM

### CODE REVIEW

File: server/models/user_model.py [hunk 1]
```diff
@@ -12,6 +12,10 @@ class UserModel(db.Model):
     full_name = db.Column(db.String(50), nullable=False)
     email = db.Column(db.String(120), unique=True, nullable=False)
     password = db.Column(db.String(120), nullable=False)
+    experience = db.Column(db.JSON, nullable=True)
```
Comment (ko1keith) [inline server/models/user_model.py]: This line is no longer needed as experience has been moved to its own model.

File: server/models/user_model.py [hunk 2]
```diff
@@ -12,6 +12,10 @@ class UserModel(db.Model):
...
+    reviews = db.Column(db.Integer, nullable=False)
```
Comment (ko1keith) [inline server/models/user_model.py]: reviews column has also been removed as its been moved to its own model.

File: client/src/pages/Profile.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+          />
+          <img src={image} className={classes.profileImage} />
+          <Typography className={classes.profileName}>{name}</Typography>
+          <Typography className={classes.title}>
+            Senior Developer at Google
+          </Typography>
+          <div className={classes.statsContainer}>
+            <Stat name={"years of experience"} number={5} classes={classes} />
+            <Stat name={"reviews"} number={24} classes={classes} />
+            <Stat name={"raiting"} number={0.8} classes={classes} />
```
Comment (ko1keith) [inline client/src/pages/Profile.js]: I think you meant rating here

File: client/src/components/ExperiencePicker.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  const availableLanguages = Object.keys(userExperience).filter((lang) => {
+    return rows[lang] == null;
+  });
+
+  const upadateRow = (rowObj) => {
+    setRows({ ...rows, ...rowObj });
+  };
+  const isExperienceEmpty = (rows) => {
+    let isEmpty = true;
+    for (var key in rows) {
```
Comment (abhishekchohan) [inline client/src/components/ExperiencePicker.js]: Try using **let** instead of **var** here

**Review by abhishekchohan — APPROVED** [review]
Seems good except for one small thing. Keith pointed out a few typos and things that need to be taken care of. One more thing, don't forget to remove console.log statements at various places if not required anymore. Other than that 👍

File: client/src/App.js [hunk 5]
```diff
@@ -81,16 +82,31 @@ const DefaultStack = () => {
...
 
 const AuthStack = ({ experience }) => {
+  const hasExperience = (experience) => {
+    let ans = false;
+    Object.values(experience).forEach((value) => {
+      if (value != null) {
+        ans = true;
+      }
+    });
+    console.log(ans);
```
Comment (skassam21) [inline client/src/App.js]: remove the console.log

File: client/src/components/ExperiencePicker.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+const experienceOptions = {
+  Beginner: 0,
+  Junior: 1,
+  Intermediate: 2,
+  Senior: 3,
+};
+
+const experienceLabelList = Object.keys(experienceOptions);
+const uploadUserExperience = (experiences, dispatch) => {
+  let outObj = {};
```
Comment (skassam21) [inline client/src/components/ExperiencePicker.js]: nitpick - this should be const

**Review by skassam21 — APPROVED** [review]
really small things

**Pull request merged**